#37 Change acr-sp-init.sh to adhere to naming convention

rickrain opened the pull request:
This PR includes a new script, `./setup/acr-setup-test.sh` to automate testing the functionality of `./setup/acr-setup.sh`.  To test the new functionality, run the following commands from a bash prompt:

``` bash
# Authenticate to Azure
az login   

cd ./tests
./acr-sp-init-test.sh
```

manojvazirani commented:
@rickrain: I was reviewing this PR and trying to execute the script but got the following error:
**$** bash --version
GNU bash, version 5.0.3(1)-release (x86_64-apple-darwin18.2.0)
Copyright (C) 2019 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

**$** acr-sp-init.sh -a test -l eastus -s test123
**./acr-sp-init.sh: line 34: ${appname,,}: bad substitution**

rickrain commented:
@xtophs The title of the PR is the same as the issue #37.  The scope of the changes in this PR are exactly what was defined in the AC with one exception, and that is the logic I added to retry role assignments instead of delaying for 20s which we talked about.  So, added that enhancement while I was making the other changes.

Title has been updated to reference #37.  The individual commits referenced #37 already but I also tagged it in the title.
